api-docs/openapi/plugins/decorators/tags/set-tag-groups.js

```diff
@@ -19,17 +19,56 @@ function SetTagGroups(data) {
     data = [];
   }
 
+  const ALL_ENDPOINTS = 'All endpoints';
+  /** Tag names used for ad-hoc grouping of Operations and not specific to a resource or path.
+   *  For example, these might be useful for UI navigation and filtering, but shouldn't appear
+   *  in a list of resource tags.
+  */
+  const nonResourceTags = [
+  'Data I/O endpoints',
+  'Security and access endpoints',
+  'System information endpoints'
+  ];
+
+  const allEndpointsGroup = data.filter(customGroup => customGroup.name === ALL_ENDPOINTS).pop();
+
+  function addAllEndpointTags(tagGroups) {
+    tagGroups.map(grp => {         
+      if(grp.name === ALL_ENDPOINTS && !grp.tags.length) {
+        grp.tags = endpointTags;
+      }
+      return grp;
+    })
+  }
+
   let tags = [];
   /** Collect tags for each operation and convert string tags to object tags. **/
   return {
     DefinitionRoot: {
       Operation: {
-        leave(op) {
+        leave(op, ctx) {
           let opTags = op.tags?.map(
             function(t) {
               return typeof t === 'string' ? { name: t } : t;
             }
           ) || [];
+
+          const { parent, key } = ctx;
+          if(allEndpointsGroup?.tags.length) {
```

Comment:
Fixes unintended deletion of API operations--only delete missing tags if allEndpointsGroup contains tags.

Reply:
> Fixes unintended deletion of API operations--only delete missing tags if allEndpointsGroup contains tags.

I thought I tested this multiple times across OSS, Cloud, IOx, and v1. Maybe it didn't refresh when I thought it did.